src/components/Calendar.tsx

```diff
@@ -0,0 +1,101 @@
+import { Badge, Calendar } from 'antd';
+import React from 'react';
+import { Link } from 'react-router-dom';
+import styled from 'styled-components';
+import { EventProps } from '../containers/upcoming-events/ducks/types';
+import { LINK } from '../utils/colors';
+
+const EventsCalendarList = styled.ul`
+  margin: 0;
+  padding: 0;
+  list-style: none;
+  .ant-badge-status {
+    width: 100%;
+    overflow: hidden;
+    font-size: 12px;
+    white-space: nowrap;
+    text-overflow: ellipsis;
+  }
+  .ant-badge-status-text {
+    transition: color 300ms;
+  }
+  .ant-badge-status-text :hover {
+    color: ${LINK};
+  }
+`;
+
+interface CalendarData {
+  eventId: number;
+  type: 'success' | 'warning' | 'error' | 'default' | 'processing' | undefined;
+  content: string;
+}
+
+interface CalendarProps {
+  events: EventProps[];
+}
+
+const matchesDateMonthYear = (eventDate: Date, value: any): boolean => {
```

Comment:
We should add a type for any here

Reply:
Ah yeah these are moment objects, I think we might be able to just assign a relevant interface for them though.